playground/infrastructure/test_helper.py

```diff
@@ -13,12 +13,163 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-from api.v1.api_pb2 import SDK_UNSPECIFIED, STATUS_UNSPECIFIED
-from helper import find_examples, Example
+import mock
 
+from unittest.mock import mock_open
+from api.v1.api_pb2 import SDK_UNSPECIFIED, STATUS_UNSPECIFIED, SDK_JAVA
+from helper import find_examples, Example, _get_example, _match_pattern, _get_name, _get_sdk, get_tag, _validate
 
-def test_find_examples():
-    result = find_examples("")
 
-    assert len(result) == 1
-    assert result[0] == Example("", "", SDK_UNSPECIFIED, "", "", "", STATUS_UNSPECIFIED)
+@mock.patch('helper._get_example')
+@mock.patch('helper._match_pattern')
+@mock.patch('helper.os.path.join')
+@mock.patch('helper.os.walk')
+def test_find_examples(mock_os_walk, mock_os_path_join, mock_match_pattern, mock_get_example):
+    example = Example("file", "pipeline_id", SDK_UNSPECIFIED, "root/file.extension", "code", "output",
+                      STATUS_UNSPECIFIED)
+    mock_os_walk.return_value = [("/root", (), ("file.extension",))]
+    mock_os_path_join.return_value = "/root/file.extension"
+    mock_match_pattern.return_value = True
+    mock_get_example.return_value = example
+
+    find_examples("")
+
+    mock_os_walk.assert_called_once_with("")
+    mock_os_path_join.assert_any_call("/root", "file.extension")
+    mock_match_pattern.assert_called_once_with("/root/file.extension")
+    mock_get_example.assert_called_once_with("/root/file.extension", "file.extension")
+
+
+@mock.patch('builtins.open',
+            mock_open(
+                read_data="...\n# limitations under the License.\n# Beam-playground:\n#     name: Name\n\nimport ..."))
+def test_get_tag_when_tag_is_exists():
+    result = get_tag("")
+
+    assert result.get("name") == "Name"
+
+
+@mock.patch('builtins.open',
+            mock_open(
+                read_data="...\n# limitations under the License.\n\nimport ..."))
+def test_get_tag_when_tag_does_not_exist():
+    result = get_tag("")
+
+    assert result is None
+
+
+@mock.patch('builtins.open',
+            mock_open(
+                read_data="...\n# limitations under the License.\n# Beam-playground:\n# additional string\n#     name: Name\n\nimport ..."))
+def test_get_tag_when_tag_has_incorrect_format():
+    try:
+        get_tag("")
+        assert False
+    except Exception:
+        assert True
```

Comment:
In tests, it is better to use the constructions described [here](https://docs.pytest.org/en/6.2.x/assert.html#assertions-about-expected-exceptions). And here is also best practice to expect the exact exception, not the generic one.